File: jna/src/main/java/dark/ious.java
package dark;

import com.sun.jna.Pointer;
import com.sun.jna.Structure;

import java.util.Arrays;
import java.util.List;

public class ious extends Structure {
    public float iou, giou, diou, ciou;
    public dxrep dx_iou;
    public dxrep dx_giou;

    public ious() {
        super();
    }

    public ious(Pointer peer) {
        super(peer);
    }

    @Override
    protected List<String> getFieldOrder() {
        return Arrays.asList("iou", "giou", "diou", "ciou", "dx_iou", "dxrep", "dx_giou");
    }
}

File: jna/src/main/java/dark_kotlin/ious.kt
package com.sl.microservice.ai.runtime.impl.darknet.dark

import com.sun.jna.Pointer
import com.sun.jna.Structure

class ious : Structure {
    @JvmField var iou = 0f
    @JvmField var giou = 0f
    @JvmField var diou = 0f
    @JvmField var ciou = 0f
    @JvmField var dx_iou: dxrep? = null
    @JvmField var dx_giou: dxrep? = null

    constructor() : super()
    constructor(peer: Pointer) : super(peer) {
        read()
    }

    override fun getFieldOrder(): List<String> {
        return listOf("iou", "giou", "diou", "ciou", "dx_iou", "dxrep", "dx_giou")
    }
}
